instrumentation/prometheus.go

```diff
@@ -0,0 +1,62 @@
+package instrumentation
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/caraml-dev/mlp/api/log"
+	"github.com/caraml-dev/mlp/api/pkg/instrumentation/metrics"
+)
+
+const (
+	// Namespace is the Prometheus Namespace in all metrics published by the Webhook app
+	Namespace string = "flyte"
+	// Subsystem is the Prometheus Subsystem in all metrics published by the Webhook app
+	Subsystem string = "dsw"
+)
+
+const (
+	MLPSecretsNotFound   metrics.MetricName = "mlp_secrets_not_found"
+	MLPRequestsTotal     metrics.MetricName = "mlp_requests_total"
+	WebhookRequestsTotal metrics.MetricName = "webhook_requests_total"
+)
+
+func GetCounterMetrics() map[metrics.MetricName]metrics.PrometheusCounterVec {
+
+	var counterMap = map[metrics.MetricName]metrics.PrometheusCounterVec{
```

Comment:
nit: I feel that this way of declaring metrics is a bit complicated. Typically, I would just declare the metrics as a `var` so that it doesn't required separate initialization and use the built-in method provided by the metrics (ie. .Inc() for counter) e.g.:
```
var MLPSecretsNotFound = prometheus.NewCounterVec(prometheus.CounterOpts{
    Namespace:   Namespace,
    Subsystem:   Subsystem,
    Name:        string(MLPSecretsNotFound),
    Help:        "Number of occurrence where user requested secrets is not found in MLP API",
    ConstLabels: nil,
},
    []string{"project"},
)


func useMetrics() {
    MLPSecretsNotFound.With(labels).Inc()
```

Don't have strong opinion on this, but I feel it violates KISS principle.

Reply:
> way of declaring metrics is a bit complicated
do you mean not using `github.com/caraml-dev/mlp/api/pkg/instrumentation/metrics.InitPrometheusMetricsCollector()`, and simply create a `prometheus.NewCounter`  directly and its `.Inc()` ?

Reply:
oh wait the way it is init via `GetCounterMetrics()`, ok will look into it

Reply:
did bit of readup, there is `promauto` which can register during init
`promauto.NewCounterVec(...)` 

modified to use this and use `WithLabelValues` which is neater